### PR TITLE
First take on cursor position

### PR DESCRIPTION
**Pull request by thomasballinger**
This is an in-progress PR so there's something to talk about while I make progress. Things I can think of to worry about:

* How should this be tested?
* Anything different to do for signals received during this blocking call?
* https://github.com/erikrose/blessings/issues/65 says Sun needs to be treated differently, and that this won't work in Windows. What's the right way to detect these situations?

I'll keep working on these but would appreciate thoughts.

### CODE REVIEW

**Comment by jquast** [comment]
Well there is plenty of time, now.  I'd really like to dive into and help with your curtsies project (i've been watching and seeing your issues/pr's... i see what you're doing, but no comment) I'm currently **focused on getting blessed merged back into blessings**. 

The blessed project will become very little code: only "imports blessings, monkey patch the previous behavior/methods for compatibility, recommend blessings to be used instead".

Erik transferred blessed code into his project as a branch we're working through some weeks ago, so in effect, **"blessed" is in feature freeze**. I proposed we get this specific feature in before the transfer, but erik declined it. **This feature will have to be a part of some later release of blessings.**

I will transfer this and other tickets after the merge.

As you're looking at using blessed for keyboard, you should know that I fought a tough battle over some of the api's and won some and lost some. Erik declined the ``inkey``, ``cbreak`` and ``raw`` methods, preferring instead to rename them to ``keystroke``, and ``keystroke_input(raw=False)``, there may be even more in the keyboard.py module, for a time he renamed the ``Keystroke`` class to ``Key`` but I think I got that reversed.  There is some danger that it won't be directly compatible with the underlying code.

(It's too bad, really I asked for his feedback almost 2 years ago on these methods and their names before blessed was born, and he didn't have any opinion to share then! I have over **267** uses of inkey currently across projects, and that's only me!)

**Comment by jquast** [comment]
Also, Good luck with your pycon talk!

**Comment by thomasballinger** [comment]
Thanks for the updates, and great work on doing this merge with Eric. Thanks for the great feedback on the PR a while ago, sorry I didn't make any progress on it after you gave great specific feedback.

File: blessed/terminal.py [hunk 1]
```diff
@@ -280,6 +281,27 @@ def width(self):
...
+            first = 0
+            keystrokes = []
+            while True:
+                keystrokes.append(self.inkey(_intr_continue=True))
+                m = re.search('\x1b[[]([0-9]+);([0-9]+)R', u''.join(keystrokes))
+                if m:
+                    self._keyboard_buf.extendleft(
+                            u''.join(keystrokes)[:m.start()])
+                    return tuple(int(x) for x in m.groups())
+
```
Comment (jquast) [inline blessed/terminal.py]: man... lot of mojo here without any test cases !

File: blessed/terminal.py [hunk 2]
```diff
@@ -280,6 +281,27 @@ def width(self):
...
+            self.stream.write(CURSOR_QUERY)
+            self.stream.flush()
+
+            first = 0
+            keystrokes = []
+            while True:
+                keystrokes.append(self.inkey(_intr_continue=True))
+                m = re.search('\x1b[[]([0-9]+);([0-9]+)R', u''.join(keystrokes))
+                if m:
+                    self._keyboard_buf.extendleft(
```
Comment (jquast) [inline blessed/terminal.py]: this may warrant the need for a _ungetch() method.

**Comment by jquast** [comment]
resuming

**Comment by jquast** [comment]
This PR superseded by #72

**Pull request closed without merging**